aries_cloudagent/askar/profile.py

```diff
@@ -146,6 +153,24 @@ async def close(self):
             await self.opened.close()
             self.opened = None
 
+    def finalizer(self) -> Optional[finalize]:
+        """Return a finalizer for this profile.
+
+        See docs for weakref.finalize for more details on behavior of finalizers.
+        """
+        # Askar Profiles (not to be confused with AskarProfiles) don't need
+        # additional clean up
+
+        if self.profile_id:
+            return None
+
+        def _finalize(opened: Optional[AskarOpenStore]):
+            if opened:
+                LOGGER.debug("Profile finalizer called; closing wallet")
+                asyncio.get_event_loop().create_task(opened.close())
```

Comment:
It's nice but not necessary to close the store this way, as the library has its own finalizers. With multiple profile instances potentially referencing the same Store it would probably not be correct to close the store when a single profile is closed, either.